File: NET/API/Treehopper/SpiMode.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Treehopper
{
    /// <summary>
    /// Defines the clock phase and polarity used for SPI communication
    /// </summary>
    /// <remarks>
    /// <para>The left number indicates the clock polarity (CPOL), while the right number indicates the clock phase (CPHA). Consult https://en.wikipedia.org/wiki/Serial_Peripheral_Interface_Bus#Clock_polarity_and_phase for more information.</para>
    /// <para>Note that the numeric values of this enum do not match the standard nomenclature, but instead match the value needed by Treehopper's MCU. Do not attempt to cast integers from/to this enum.</para>
    /// </remarks>
    public enum SpiMode
    {
        /// <summary>
        /// Clock is initially low; data is valid on the rising edge of the clock
        /// </summary>
        Mode00 = 0x00,

        /// <summary>
        /// Clock is initially low; data is valid on the falling edge of the clock
        /// </summary>
        Mode01 = 0x20,

        /// <summary>
        /// Clock is initially high; data is valid on the rising edge of the clock
        /// </summary>
        Mode10 = 0x10,

        /// <summary>
        /// Clock is initially high; data is valid on the falling edge of the clock
        /// </summary>
        Mode11 = 0x30
    };
}

File: Java/api/treehopper/src/main/java/io/treehopper/enums/SpiMode.java
package io.treehopper.enums;

/**
 * Defines the clock phase and polarity used by the SPI peripheral
 */
public enum SpiMode {
    /**
     * Clock is initially low; data is valid on the rising edge of the clock
     */
    Mode00((byte) 0x00),

    /**
     * Clock is initially low; data is valid on the falling edge of the clock
     */
    Mode01((byte) 0x20),

    /**
     * Clock is initially high; data is valid on the rising edge of the clock
     */
    Mode10((byte) 0x10),

    /**
     * Clock is initially high; data is valid on the falling edge of the clock
     */
    Mode11((byte) 0x30);

    private final byte modeRegisterValue;

    SpiMode(byte modeRegisterValue) {
        this.modeRegisterValue = modeRegisterValue;
    }

    public byte getModeRegisterValue() {
        return modeRegisterValue;
    }
}